src/mixer.c

```diff
@@ -506,33 +506,85 @@ int mixer_wait_event(struct mixer *mixer, int timeout)
 #endif
 
     if (!count)
-        return 0;
+        goto exit;
 
     for (;;) {
         int err;
         err = poll(pfd, count, timeout);
-        if (err < 0)
-            return -errno;
+        if (err < 0) {
+            ret = -errno;
+            goto exit;
+        }
         if (!err)
-            return 0;
+            goto exit;
+
         for (i = 0; i < count; i++) {
-            if (pfd[i].revents & (POLLERR | POLLNVAL))
-                return -EIO;
+            if (pfd[i].revents & (POLLERR | POLLNVAL)) {
+                ret = -EIO;
+                goto exit;
+            }
             if (pfd[i].revents & (POLLIN | POLLOUT)) {
                 if ((i == 0) && mixer->fd >= 0) {
                     grp = mixer->h_grp;
                     grp->event_cnt++;
                 }
 #ifdef TINYALSA_USES_PLUGINS
-				else {
+                 else {
                     grp = mixer->v_grp;
                     grp->event_cnt++;
                 }
 #endif
-                return 1;
+                ret = 1;
+                goto exit;
```

Comment:
I don't want to keep adding goto statements, please try to avoid them in future patches.